### PR TITLE
Memory allocation audit

### PR DESCRIPTION
**Pull request by pks-t**
Various fixes to potential memory allocation errors which were previously unchecked. Triggered by #5199 and #5164, even though I've got no idea whether it will fix these reported issues.

### CODE REVIEW

File: src/posix.c [hunk 1]
```diff
@@ -28,7 +28,7 @@ int p_getaddrinfo(
 
 	GIT_UNUSED(hints);
 
-	if ((ainfo = malloc(sizeof(struct addrinfo))) == NULL)
+	if ((ainfo = git__malloc(sizeof(struct addrinfo))) == NULL)
```
Comment (ethomson) [inline src/posix.c]: we'll need to `p_free` this in cleanup (line 35)

File: src/posix.c [hunk 2]
```diff
@@ -65,7 +65,7 @@ int p_getaddrinfo(
 	ai = ainfo;
 
 	for (p = 1; ainfo->ai_hostent->h_addr_list[p] != NULL; p++) {
-		if (!(ai->ai_next = malloc(sizeof(struct addrinfo)))) {
+		if (!(ai->ai_next = git__malloc(sizeof(struct addrinfo)))) {
```
Comment (ethomson) [inline src/posix.c]: We'll need to `p_free` this in `p_freeaddrinfo`

File: src/posix.c [hunk 3]
```diff
@@ -247,7 +247,7 @@ int p_mmap(git_map *out, size_t len, int prot, int flags, int fd, git_off_t offs
 		return -1;
 	}
 
-	out->data = malloc(len);
+	out->data = git__malloc(len);
```
Comment (ethomson) [inline src/posix.c]: Need to `p_free` this in `p_unmap`

**Comment by pks-t** [comment]
Thanks, good catch @ethomson! Fixed the missing free conversions

**Pull request merged**